src/components/Common/EmojiPickerIconButton.tsx

```diff
@@ -0,0 +1,74 @@
+import { Box, IconButton } from "@mui/material";
+import EmojiEmotionsOutlinedIcon from "@mui/icons-material/EmojiEmotionsOutlined";
+import { useState, useEffect, useRef } from "react";
+import EmojiPicker, { EmojiClickData, EmojiStyle } from "emoji-picker-react";
+
+type EmojiPickerIconButtonProps = {
+  onEmojiClick: (emoji: EmojiClickData) => void;
+  pickerHeight?: number;
+  pickerPosition?: "top" | "bottom";
+  pickerWidth?: number;
+};
+
+const styles = {
+  emojiContainer: { position: "absolute", zIndex: 1 },
+  emojiPicker: {
+    "--epr-emoji-size": "25px",
+  },
+};
+
+const EmojiPickerIconButton = ({
+  onEmojiClick,
+  pickerHeight = 450,
+  pickerPosition = "bottom",
+  pickerWidth = 350,
+}: EmojiPickerIconButtonProps) => {
+  const emojiContainerRef = useRef<HTMLDivElement>(null);
+  const emojiIconButtonRef = useRef<HTMLButtonElement>(null);
+  const twitterEmojiStyle = EmojiStyle.TWITTER;
+  const [showEmojiPicker, setShowEmojiPicker] = useState(false);
+  const top = pickerPosition === "bottom" ? 150 : -460;
+
```

Comment:
If we follow my suggestion, top position would need to go down by about 100..

```suggestion
  const top = topPosition ? -360 : 150;
```

Reply:
This isn't the ideal solution as we are hardcoding the numbers, but I think I want to prioritize shipping this part and worrying about dynamic pages at a later date

Reply:
I changed the code as follows: 

`const top = topPosition ? -(pickerHeight + 10) : 150;`

I think this is fine because this way the picker will always be 10 pixels above the button if it's positioned on top. And if it's positioned on the bottom I think hard coding it to 150 will always make it 10 pixels below the button (unless the height of the picker is <150, but that would be a terribly small emoji picker)

Reply:
Yeah thats fair enough, good stuff dennis